.github/workflows/markdown-links-check/markdown-links-check-config.json

```diff
@@ -0,0 +1,5 @@
+{
+  "timeout": "5s",
+  "retryCount":10,
+  "aliveStatusCodes": [0, 200, 302, 400, 403, 429]
```

Comment:
Error code 400 is a bad request which does not seem appropriate to consider a valid link.  From the logs, what URI is triggering that status code?

Also allowing 429 seems like it could lead to some false-positives, I'd prefer we try the `retryOn429` and `retry-after` settings to work around these error codes, otherwise we could have a bad link that the server kicks back as 429 and we think it's valid.

I'm also curious which URIs are triggering error code 403, as that implies some of our links require authentication.  I assume all of our links reference public resources, but maybe there are some exceptions.

0 is not a valid HTTP status code, so I'm also curious which URIs trigger that response.

Reply:
For code 400, it could be found in some changelog.md file like:
![image](https://user-images.githubusercontent.com/84758614/158374271-e2ad94d7-fd17-4b4f-bb71-8f72b10c4149.png)
For code 429, all github issues links will be marked 429 because it send requests too much and fast, see:
![image](https://user-images.githubusercontent.com/84758614/158374800-2b291d94-7d5d-4685-bd3c-4d4a748042e9.png)
For code 403, it seems the github action runner(docker) cannot reach maven repository, see:
![image](https://user-images.githubusercontent.com/84758614/158374972-965eb6e1-12d1-4342-8cb0-1fd6120c43aa.png)
For code 0, it could be found in getting-started-alluxio.md, it is a cluster url, not really need to link, see:
![image](https://user-images.githubusercontent.com/84758614/158375550-f446dce0-83f8-4dff-9da8-4a3730597d73.png)
all logs can be found [here](https://github.com/nvliyuan/yuali-spark-rapids/actions)

Reply:
for status code 429, I also see other open source repo add it as an exception, see
https://github.com/netdata/netdata/blob/master/.mlc_config.json and retryon429 is also a workaround see
https://github.com/hashicorp/packer/blob/main/.mlc_config.json

Reply:
I'm not saying we can just ignore 429, but if `retryon429` (and possibly a `retry-after` setting to go along with it) works, that is far preferable to just taking any 429 and counting it as success.

Reply:
Added `retryon429`.

Exclude pattern `https://github.com/NVIDIA/spark-rapids/pull/*` and `https://github.com/NVIDIA/spark-rapids/issues/*` because there are thousands of rapids PRs and issues URL links we don't really need to check, and this will spend a long time to check, and it cannot be avoid by `check-modified-files-only` because these thousands of links are in CHANGELOG file.

Exclude pattern `/docs/archives`, this relative link can only be found in CHANGELOG, and the link is not broken, please see https://github.com/nvliyuan/yuali-spark-rapids/blob/branch-22.04-add-markdown-action/CHANGELOG.md   I don't know why the link checker mark it as broken, maybe a bug?

Remove status code 0(status code 0 is defined by the link checker, if we ignore some links, these links will be mark as status 0)